zapr_test.go

```diff
@@ -164,6 +198,33 @@ func TestInfo(t *testing.T) {
 `,
 			keysValues: []interface{}{"duration", time.Duration(5 * time.Second)},
 		},
+		{
+			msg: "valid marshaler",
+			format: `{"ts":%f,"caller":"zapr/zapr_test.go:%d","msg":"valid marshaler","v":0,"obj":"msg=hello"}
+`,
+			keysValues: []interface{}{"obj", &marshaler{msg: "hello"}},
+		},
+		{
+			msg: "nil marshaler",
+			// Handled by our code: it just formats the error.
+			format: `{"ts":%f,"caller":"zapr/zapr_test.go:%d","msg":"nil marshaler","v":0,"obj":"<panic: runtime error: invalid memory address or nil pointer dereference>"}
```

Comment:
I'm on the fence whether this should be consistent with funcr or with zap (see below).

Reply:
I would make it consistent with the underlying library, I think.  I don't like the pattern, but I can see the argument.  IMO the user chose to use zap, they should experience zap's design decisions (for better or worse)

Reply:
Okay. I've updated the handling accordingly.